### PR TITLE
Added missing fix its

### PR DESCRIPTION
**Pull request by kimdv**
Not sure if this is needed, but I tought it might be a good idea to compare the fixits we expect and what we actually get.

So just got some examples of where it was missing.
There is still a few hundred places missing.

Also tought that I adding `fixedSource` so we ensure they works in as many scenarios as possible.

### CODE REVIEW

File: Tests/SwiftParserTest/Assertions.swift [hunk 1]
```diff
@@ -594,6 +594,7 @@ func assertParse<S: SyntaxProtocol>(
     )
   } else {
     for (diag, expectedDiag) in zip(diags, expectedDiagnostics) {
+      XCTAssertEqual(diag.fixIts.map { $0.message.message }, expectedDiag.fixIts ?? [], file: file, line: expectedDiag.line)
```
Comment (ahoppen) [inline Tests/SwiftParserTest/Assertions.swift:597]: We should be checking this exact same thing in
https://github.com/apple/swift-syntax/blob/05356239a2b5b3f5c54653a70ec47cb17d5afd14/Tests/SwiftParserTest/Assertions.swift#L437-L446

**Comment by kimdv** [comment]
An assertion message could look like `testSwitch3(): XCTAssertEqual failed: ("["insert \'{}\'"]") is not equal to ("[]")`
Should that be improved ?

File: Tests/SwiftParserTest/Assertions.swift (hunk 1 above)
Comment (kimdv) [reply Tests/SwiftParserTest/Assertions.swift:597]: Changed the logic a bit.

**Comment by ahoppen** [comment]
I think that’s good enough but if you want to improve it, I won’t stop you.

**Comment by kimdv** [comment]
@ahoppen we need a bit of planning when merging this. 
When it's approved I will rebase on latest main and trigger CI right after. 

I think we then should wait a bit on merging other things, to avoid test start failing because of missing fix-its.

**Comment by kimdv** [comment]
@swift-ci please test Windows

File: Tests/SwiftParserTest/Assertions.swift [hunk 2]
```diff
@@ -434,7 +434,18 @@ func assertDiagnostic<T: SyntaxProtocol>(
       }
     }
   }
-  if let fixIts = spec.fixIts {
+
+  if diag.fixIts.count != (spec.fixIts?.count ?? 0) {
```
Comment (ahoppen) [inline Tests/SwiftParserTest/Assertions.swift:438]: Should we just make `spec.fixIts` non-optional and default to `[]`? That way we don’t need to handle with optionals here.

**Review by ahoppen — APPROVED** [review]
Thank you. Adding all those expected Fix-Its must have taken you some time 😆 It’s a great change to better testing. I like it. 

One minor comment, otherwise you can try landing it over the weekend.

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by kimdv** [comment]
@swift-ci please test windows

**Pull request merged**